src/ansys/geometry/core/connection/product_instance.py

```diff
@@ -298,6 +303,22 @@ def get_available_port():
     return port
 
 
+def _wait_for_backend(host: str, port: int, timeout: int):
+    """Check if the backend is ready to accept connections."""
+    import time
+
+    start_time = time.time()
+    while time.time() - start_time < timeout:
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+            if s.connect_ex((host, port)) == 0:
+                return
```

Comment:
A connection message would be nice here
```suggestion
        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
            if s.connect_ex((host, port)) == 0:
                LOG.info("Successful connection to backend.")
                return
```

Reply:
Yep agreed - but let me modify the message. Connection is not established. We are just verifying that the port is occupied by the service